### PR TITLE
[feature]: added docker command

### LINKED ISSUE #1505: [feature] : add docker command supoort
The GitHub action uses binary to run the GitHub Action due to which `docker command` can't be used to run the application with keploy.

```yml
- name: Keploy Report
      uses: keploy/testgpt@main
      with:
        working-directory: ./
        delay: 10
        command: docker compose up
        keploy-path: ./
```
the above config results to error below: -

image

### PR DESCRIPTION
**Pull request by Yaxhveer**
closes: keploy/keploy#1505
Added support for docker and docker compose.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -42,6 +42,18 @@ elif [[ "$COMMAND" =~ .*"python".* ]] || [[ "$COMMAND" =~ .*"python3".* ]]; then
   echo sudo -E keploy test -c "${COMMAND}" --delay ${DELAY} --path "${KEPLOY_PATH}"
   sudo -E keploy test -c "${COMMAND}" --delay ${DELAY} --path "${KEPLOY_PATH}"
 
+elif [[ "$COMMAND" =~ .*"docker-compose".* ]] || [[ "$COMMAND" =~ .*"docker compose".* ]]; then
+  echo "Docker compose is present."
+  echo 'Test Mode Starting 🎉'
+  echo sudo -E keploy test -c "${COMMAND}" --delay ${DELAY} --path "${KEPLOY_PATH}" --containerName "${CONTAINER_NAME}"
+  sudo -E keploy test -c "${COMMAND}" --delay ${DELAY} --path "${KEPLOY_PATH}" --containerName "${CONTAINER_NAME}"
```
Comment (Sonichigo) [inline install.sh]: In case of Docker command usage `--delay` would be replaced by `--buildDelay`.

File: install.sh [hunk 2]
```diff
@@ -42,6 +42,18 @@ elif [[ "$COMMAND" =~ .*"python".* ]] || [[ "$COMMAND" =~ .*"python3".* ]]; then
...
+
+elif [[ "$COMMAND" =~ .*"docker".* ]]; then
+  echo "Docker is present."
+  echo 'Test Mode Starting 🎉'
+  echo sudo -E keploy test -c "${COMMAND}" --delay ${DELAY} --path "${KEPLOY_PATH}"
+  sudo -E keploy test -c "${COMMAND}" --delay ${DELAY} --path "${KEPLOY_PATH}"
```
Comment (Sonichigo) [inline install.sh]: same here.

**Review by Sonichigo — CHANGES_REQUESTED** [review]

File: action.yml [hunk 3]
```diff
@@ -20,6 +20,11 @@ inputs:
     description: Time to start application
     required: true
     default: 10
+  container-name:
+    description: Name of the container in case of "docker compose" command
+  build-delay:
+    description: Time to wait for docker container build
+    default: 30
```
Comment (Sonichigo) [inline action.yml]: this will result to syntax errors as no unit is mentioned! you can change it to `50s`

**Review by Sonichigo — CHANGES_REQUESTED** [review]

File: action.yml (hunk 3 above)
Comment (Yaxhveer) [reply action.yml]: Ok

Comment (Yaxhveer) [reply action.yml:27]: should i also add unit in delay too i.e, ```10s```

**Review by Sonichigo — APPROVED** [review]

**Pull request merged**